File: apps/backend/models/Warehouse.js
const mongoose = require('mongoose');

const warehouseSchema = new mongoose.Schema({
    name: {
        type: String,
        required: [true, 'Warehouse name is required'],
        trim: true,
        unique: true
    },
    code: {
        type: String,
        required: [true, 'Warehouse code is required'],
        trim: true,
        uppercase: true,
        unique: true
    },
    address: {
        addressLine: {
            type: String,
            required: [true, 'Address line is required'],
            trim: true
        },
        city: {
            type: String,
            required: [true, 'City is required'],
            trim: true
        },
        state: {
            type: String,
            trim: true
        },
        country: {
            type: String,
            required: [true, 'Country is required'],
            trim: true
        },
        zip: {
            type: String,
            required: [true, 'ZIP code is required'],
            trim: true
        }
    },
    contact: {
        name: {
            type: String,
            trim: true
        },
        phone: {
            type: String,
            trim: true
        },
        email: {
            type: String,
            trim: true,
            lowercase: true
        }
    },
    isActive: {
        type: Boolean,
        default: true
    },
    createdAt: {
        type: Date,
        default: Date.now
    },
    updatedAt: {
        type: Date,
        default: Date.now
    }
}, {
    timestamps: true
});

module.exports = mongoose.model('Warehouse', warehouseSchema);

File: apps/client_app/lib/src/features/shipment/domain/warehouse.dart
class Warehouse {
  final String id;
  final String name;
  final String code;
  final WarehouseAddress address;
  final bool isActive;

  const Warehouse({
    required this.id,
    required this.name,
    required this.code,
    required this.address,
    this.isActive = true,
  });

  factory Warehouse.fromJson(Map<String, dynamic> json) {
    return Warehouse(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      code: json['code'] ?? '',
      address: WarehouseAddress.fromJson(json['address'] ?? {}),
      isActive: json['isActive'] ?? true,
    );
  }

  // @override
  // List<Object?> get props => [id, name, code, address, isActive];
}

class WarehouseAddress {
  final String addressLine;
  final String city;
  final String state;
  final String country;
  final String zip;

  const WarehouseAddress({
    required this.addressLine,
    required this.city,
    required this.state,
    required this.country,
    required this.zip,
  });

  factory WarehouseAddress.fromJson(Map<String, dynamic> json) {
    return WarehouseAddress(
      addressLine: json['addressLine'] ?? '',
      city: json['city'] ?? '',
      state: json['state'] ?? '',
      country: json['country'] ?? '',
      zip: json['zip'] ?? '',
    );
  }

  String get fullAddress =>
      '$addressLine, $city, ${state.isNotEmpty ? '$state, ' : ''}$country $zip';

  // @override
  // List<Object?> get props => [addressLine, city, state, country, zip];
}
